docs: Update FIPS documentation to reference k8s-snap docs (#25)
canonical/metallb-rocks: docs/fips.md

# MetalLB FIPS Compliance

For comprehensive information about FIPS 140-3 compliance in Canonical Kubernetes, including how ROCKs are built with FIPS support, please refer to the [k8s-snap FIPS documentation](https://github.com/canonical/k8s-snap/blob/main/docs/dev/fips.md).

> **Note:** As of now, FRR and pebble are not built in a FIPS-compliant way. This document will be updated once they are.

## BGP Authentication and FIPS

While MetalLB allows MD5 authentication for BGP sessions, **MD5 is not FIPS-compliant** and should be avoided in FIPS mode. MD5 authentication calls will fail on a FIPS-enabled system.

## FRR Mode

> FRR mode is not FIPS-compliant. However, enabling FRR mode is not supported in Canonical Kubernetes, so this does not impact FIPS compliance for users of that Kubernetes distribution.

Reviewer: ```suggestion
FRR mode is not FIPS-compliant. However, enabling FRR mode is not supported in Canonical Kubernetes, so this does not impact FIPS compliance for our users.
```
